src/main/java/de/tum/in/www1/artemis/repository/StudentExamRepository.java

```diff
@@ -24,22 +24,41 @@
     @EntityGraph(type = LOAD, attributePaths = { "exercises" })
     Optional<StudentExam> findWithExercisesById(Long id);
 
-    @EntityGraph(type = LOAD, attributePaths = { "exercises" })
+    @Query("select distinct se from StudentExam se left join fetch se.exercises e where se.testRun = false and se.exam.id = :#{#examId} and se.user.id = :#{#userId} ")
     Optional<StudentExam> findWithExercisesByUserIdAndExamId(@Param("userId") long userId, @Param("examId") long examId);
```

Comment:
Question to the naming pattern: Shouldn't these method names also include a `IgnoreTestRuns`, like `findWithExercisesByUserIdAndExamIdIgnoreTestRuns`

Reply:
I have followed the convention of distinguishing between student exams and test runs. Repository calls which return StudentExams which are test runs follow the naming: `findTestRun`, whereas all calls named `findStudentExam`.. ignore them. 
I think this makes it clearer which function to use as we want to distinguish clearly between student exams and test runs